.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '29 11 * * 4'
```

Comment:
Why do we need schedule? 
I think we can do it for each PR.

Reply:
This is the default behavior. I believe it is to cover the case where a new vulnerability is discovered on the CodeQL side but there is not a pull request to main for some time.